packages/react-core/src/components/Modal/ModalBoxCloseButton.tsx

```diff
@@ -12,9 +13,17 @@ export interface ModalBoxCloseButtonProps {
 export const ModalBoxCloseButton: React.FunctionComponent<ModalBoxCloseButtonProps> = ({
   className = '',
   onClose = () => undefined as any,
+  ouiaId,
   ...props
 }: ModalBoxCloseButtonProps) => (
-  <Button className={className} variant="plain" onClick={onClose} aria-label="Close" {...props}>
+  <Button
+    className={className}
+    variant="plain"
+    onClick={onClose}
+    aria-label="Close"
+    {...(ouiaId && { ouiaId: `${ouiaId}-${ModalBoxCloseButton.displayName}` })}
```

Comment:
I'm not familiar with OUIAProps yet, but does this usage of `ouiaId` meet the user requirements to be able to specify a custom `data-ouia-component-id`?

Reply:
@jeffpuzzo  Yes.  I explain in the description for this  [issue](https://github.com/patternfly/patternfly-react/pull/6732#issue-1093444865) how you can see this demonstrated.  
I talked to @jgiardino and Tiffany offline and what they were actually looking for is to utilize the `ouiaId` for identification purposes. The modal is already enabled to use `ouiaId`.  Since the Modal is not compassable, the consumer can not add `ouiaId` to the close button.  I pass the model's `ouiaId` to the close button and append the close button's display name to it so it is unique then  I add the `ouiaId` to the button.
You can read more about OUIA [here](https://www.patternfly.org/v4/developer-resources/open-ui-automation)